lib/offline/storage.js

```diff
@@ -770,6 +788,61 @@ shaka.offline.Storage.prototype.createOfflineManifest_ = function(
   };
 };
 
+/**
+ * Extract DRM initData from first segment
+ *
+ * @param {shaka.extern.Manifest} manifest
+ * @param {shaka.net.NetworkingEngine} networkingEngine
+ * @return {Promise<shaka.extern.Manifest>}
```

Comment:
Why does a function about extracting init data from segments return a manifest? Since the function is about extracting init data, please have it return something that reflects that, like an array of init data. If the function is about "filling in" drm info, please name the function as such and return nothing as it would modify a given manifest.

In either case, please add documentation to the function so that these nuances are clear.

Reply:
The reason this method is requiring and returning a manifest is because we currently inject the PSSH initData directly into the manifest. As this is actioned before the DRMEngine is initialised, it is seamless and minimises the impact of potential other changes of the shaka source to support this use case (being it only applicable to offline DRM content). We could probably have been a bit more elaborate on this and are happy to update the naming conventions + docs